backend/src-common/src/main/java/org/eclipse/sw360/datahandler/db/DatabaseHandlerUtil.java

```diff
@@ -703,4 +730,76 @@ public static String trimProjectTag(String projectTag) {
         }
         return projectTag;
     }
+
+    /**
+     * Execute Runnable To Save Attachment To File System
+     */
+    private static void executeRunnableToSaveAttachmentToFileSystem(InputStream content, String userEmail,
+            String documentId, String attachmentId, String fileName) {
+        Runnable fileHandlerRunnable = prepareFileHandlerRunnable(content, userEmail, documentId, attachmentId,
+                fileName);
+        ATTACHMENT_FILE_SYSTEM_STORE_THREAD_POOL.submit(fileHandlerRunnable);
+    }
+
+    /**
+     * Prepare Runnable to save attachment , set permissions and write to file
+     */
+    private static Runnable prepareFileHandlerRunnable(InputStream content, String userEmail, String documentId,
+            String attachmentId, String fileName) {
+        return () -> {
+            try {
+                Path outputDir = Paths.get(ATTACHMENT_STORE_FILE_SYSTEM_LOCATION, userEmail, DOCUMENT_ID + documentId,
+                        ATTACHMENT_ID + attachmentId);
+                Path outputFile = Paths.get(fileName);
+                Path outputFilePath = outputDir.resolve(outputFile);
+                log.info("Preparing to store attachment in file system" + outputFilePath);
+                if (!Files.exists(outputFile)) {
+                    Set<PosixFilePermission> perms = PosixFilePermissions
+                            .fromString(ATTACHMENT_STORE_FILE_SYSTEM_PERMISSION);
+                    FileAttribute<Set<PosixFilePermission>> fileAttribute = PosixFilePermissions.asFileAttribute(perms);
+                    Files.createDirectories(outputDir, fileAttribute);
+                    Files.createFile(outputFilePath, fileAttribute);
+                    try (OutputStream os = Files.newOutputStream(outputFilePath, StandardOpenOption.WRITE)) {
+                        os.write(content.readAllBytes());
+                        os.flush();
+                        content.close();
+                    }
+                }
+            } catch (Exception exp) {
+                log.warn("Error occured while writing to a file", exp);
+            }
+        };
+    }
+
+    /**
+     * Compare and find newly added attachments to be saved to File System
+     */
+    public static void saveAttachmentInFileSystem(AttachmentConnector attachmentConnector, Set<Attachment> before,
+            Set<Attachment> after, String userEmail, String documentId) {
+        if (!IS_STORE_ATTACHMENT_TO_FILE_SYSTEM_ENABLED) {
+            log.info("store attachment to file system is disabled");
```

Comment:
I feel it should be changed to debug. Or else unnecessarily for every update operation this will be printed if the flag is off.

Reply:
changes to debug